### DISCUSSION TITLE
Placeholders ...

### DISCUSSION THREAD

**Discussion by njk0007** [bluenviron/mediamtx#1306] [Questions]
I have rtsp-simple-server (RSS) acting as a proxy, for streams that exist behind firewalls.  Streams are activated locally and sent to the server (using ffmpeg -f rtsp).  Currently, the stream is a random token (ABCDEF234..).  [I could change this if necessary]

The client webpages have a <video> element where we use HLS to load the http:/rtspsimpleserver/ABCDEF234/index.m3u

This works.
But there is an issue with timing.
If the webpage tries to load before the stream has been initiated it fails.

My question is vague because I suspect there is something I am missing or an easier way to accomplish my needs.

1) Is there a way to have invalid connections from the webpage show nothing (or a placeholder) until the stream become available?
2) Is there a way to change the underlying mapping from INPUT to OUTPUT streams on the fly?
3) Or is the only viable solution to continually poll RSS and not send the webpage the ABCDEF234 path until I see it in the /v1/paths/lists api?

5) Is there a way better way to do this?

**Comment by aler9** [comment]
Hello,

Currently the only way to do this is building your own video page and using hls.js to point to the video path `http://server-host:8888/path/index.m3u8`, as shown here:

https://github.com/aler9/rtsp-simple-server/blob/main/internal/core/hls_index.html

Nonetheless, you're making a point, the page that shows "404 not found" when a stream is not present is not pleasant and should be replaced by a page that is able to refresh itself when the stream becomes available.

I've created a feature request in order to track the feature: #1444

**Comment by xaionaro** [comment]
> https://github.com/aler9/rtsp-simple-server/blob/main/internal/core/hls_index.html

The link is broken.

**Reply by aler9** [reply]
Discussions are not an up-to-date reference, while the README and the code are.

[a comment by an automated account is not shown]
